src/main/scala/nl/knaw/dans/easy/transform/XmlTransformation.scala

```diff
@@ -22,51 +22,65 @@ import nl.knaw.dans.easy.transform.AccessRights.AccessRights
 
 object XmlTransformation {
 
+  val DEFAULT_DC_PREFIX = "dcterms"
+  val DEFAULT_DC_URI = "http://purl.org/dc/terms/"
+
   private val accessibleToRightsMap = Map(
     AccessRights.OPEN_ACCESS -> AccessibleToRights.ANONYMOUS,
     AccessRights.OPEN_ACCESS_FOR_REGISTERED_USERS -> AccessibleToRights.KNOWN,
     AccessRights.REQUEST_PERMISSION -> AccessibleToRights.RESTRICTED_REQUEST,
     AccessRights.NO_ACCESS -> AccessibleToRights.NONE)
 
   def enrichFilesXml(bagId: BagId, filesXml: Node, datasetXml: Node, downloadUrl: URI): Node = {
+    val dcNameSpace = getDcNamespace(filesXml)
+    val accessRights = getAccessRights(datasetXml)
     val rule = new RewriteRule {
       override def transform(n: Node): Seq[Node] = n match {
-        case elem: Elem if elem.label == "file" => enrichFileElement(bagId, elem, datasetXml, downloadUrl)
+        case elem: Elem if elem.label == "file" => enrichFileElement(bagId, elem, dcNameSpace, accessRights, downloadUrl)
         case _ => n
       }
     }
     new RuleTransformer(rule).transform(filesXml).head
   }
 
-  private def enrichFileElement(bagId: BagId, file: Elem, datasetXml: Node, downloadUrl: URI): Elem = {
+  private def enrichFileElement(bagId: BagId, file: Elem, dcNameSpace: String, accessRights: AccessRights, downloadUrl: URI): Elem = {
     val accessibleToRights = file \\ "accessibleToRights"
     val visibleToRights = file \\ "visibleToRights"
     var enriched = file
 
     if (accessibleToRights.isEmpty)
-      enriched = enriched.copy(child = enriched.child ++ getAccessibleToRightsElement(datasetXml))
+      enriched = enriched.copy(child = enriched.child ++ getAccessibleToRightsElement(accessRights))
 
     if (visibleToRights.isEmpty)
       enriched = enriched.copy(child = enriched.child ++ getVisibleToRightsElement())
 
-    addDownloadUrl(bagId, enriched, downloadUrl)
+    addDownloadUrl(bagId, enriched, downloadUrl, dcNameSpace)
   }
 
-  private def getAccessibleToRightsElement(datasetXml: Node): Elem = {
-    val accessRights: AccessRights = AccessRights.withName((datasetXml \ "profile" \ "accessRights").text)
+  private def getAccessibleToRightsElement(accessRights: AccessRights): Elem = {
     <accessibleToRights>{accessibleToRightsMap(accessRights)}</accessibleToRights>
   }
 
   private def getVisibleToRightsElement(): Elem = {
     <visibleToRights>{VisibleToRights.ANONYMOUS}</visibleToRights>
   }
 
-  private def addDownloadUrl(bagId: BagId, file: Elem, downloadUrl: URI) = {
+  private def addDownloadUrl(bagId: BagId, file: Elem, downloadUrl: URI, dcNameSpace: String) = {
     val escapedFilePath = Paths.get((file \ "@filepath").text).escapePath
-    file.copy(child = file.child ++ <dcterms:source>{downloadUrl.resolve(s"$bagId/$escapedFilePath")}</dcterms:source>)
+    val downloadPath = downloadUrl.resolve(s"$bagId/$escapedFilePath")
+    val sourceElement = XML.loadString(s"""<$dcNameSpace:source>$downloadPath</$dcNameSpace:source>""")
```

Comment:
can't you add the element in a more xml-ish way, where you specify the namespace, tag-name and value?

Reply:
I don't know how to parameterize the namespace in the tag except by using XML.loadString....